manifests/init.pp

```diff
@@ -1,133 +1,137 @@
+# @param disable_proxy
 class puppet_agent (
-  $arch                    = $facts['os']['architecture'],
-  $collection              = $::puppet_agent::params::collection,
-  $is_pe                   = $::puppet_agent::params::_is_pe,
-  $manage_pki_dir          = true,
-  $manage_repo             = true,
-  $package_name            = 'puppet-agent',
-  $package_version         = undef,
-  $service_names           = $::puppet_agent::params::service_names,
-  $source                  = undef,
-  $absolute_source         = undef,
-  $yum_source              = 'http://yum.puppet.com',
-  $apt_source              = 'https://apt.puppet.com',
-  $mac_source              = 'https://downloads.puppet.com',
-  $windows_source          = 'https://downloads.puppet.com',
-  $solaris_source          = 'puppet:///pe_packages',
-  $aix_source              = 'puppet:///pe_packages',
-  $use_alternate_sources   = false,
-  $alternate_pe_source     = undef,
-  Optional[Stdlib::Absolutepath] $install_dir = undef,
-  $disable_proxy           = false,
-  $proxy                   = undef,
-  $install_options         = [],
-  $skip_if_unavailable     = 'absent',
-  $msi_move_locked_files   = false,
-  $wait_for_pxp_agent_exit = undef,
-  $wait_for_puppet_run     = undef,
-  Array[Puppet_agent::Config] $config = [],
-  $version_file_path       = $facts['os']['family'] ? { 'windows' => "${facts['env_windows_installdir']}\\VERSION", default => '/opt/puppetlabs/puppet/VERSION' }
-) inherits ::puppet_agent::params {
+  String                         $arch                    = $facts['os']['architecture'],
+  String                         $collection              = $puppet_agent::params::collection,
+  Boolean                        $is_pe                   = $puppet_agent::params::_is_pe,
+  Boolean                        $manage_pki_dir          = true,
+  Boolean                        $manage_repo             = true,
+  String                         $package_name            = 'puppet-agent',
+  Optional                       $package_version         = undef,
+  Array                          $service_names           = $puppet_agent::params::service_names,
+  Optional                       $source                  = undef,
+  Optional                       $absolute_source         = undef,
+  String                         $yum_source              = 'http://yum.puppet.com',
+  String                         $apt_source              = 'https://apt.puppet.com',
+  String                         $mac_source              = 'https://downloads.puppet.com',
+  String                         $windows_source          = 'https://downloads.puppet.com',
+  String                         $solaris_source          = 'puppet:///modules/pe_packages',
+  String                         $aix_source              = 'puppet:///modules/pe_packages',
+  Boolean                        $use_alternate_sources   = false,
+  Optional                       $alternate_pe_source     = undef,
```

Comment:
As much as I like using puppet's types, are some of these a breaking change? For example, if someone was doing:

```puppet
class { 'puppet_agent':
  is_pe => "true"
```

Will puppet fail because it was passed a String and not a Boolean? Or will puppet automatically coerce?